sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -229,6 +229,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   def load(path: String): DataFrame = {
     // force invocation of `load(...varargs...)`
+    verifyPathOptionDoesNotExist()
```

Comment:
For API usage view, I don't very agree with this rule. A later `load(path)` can be seen as to overwrite previous `path` option.

But not against this, though.

Reply:
Think about `df.write.option(path).load(paths)`. The current behavior is:
1. if `paths` is just a single path, it overrides the path option.
2. otherwise, it combines the path option.

I would say the reasonable behavior is to always override the path option if `paths` is given. But this is a silent behavior change. And I think it's safer to fail if path option and `paths` are both specified.

Reply:
Do we consider to put a line in migration guide? This sounds a breaking change.

Reply:
Updated the migration guide.